safe-global/safe-wallet-web
# feat: Subaccounts

## What it solves

Implementation of creating (and funding) Subaccounts

## How this PR fixes it

A new sidebar icon, context menu (accessible via the icon/drawer) and settings have been added that both list and open a Subaccount creation flow.

## How to test it

Open a Safe and observe:

- A new header icon in the sidebar that, when clicked, opens either a promo message (if no Subaccounts are owned) or a list, with an option to add more. (The presence of Subaccounts is signified by a badge number.)
- If the **current** Safe has Subaccounts, a new item in the sidebar drawer menu (accessible via the three dots) should open the same component.
- A new settings area under "Setup" has the option to add Subaccounts, and list/rename those which exist already.

## Screenshots

![image](https://github.com/user-attachments/assets/acd20eb6-13f1-4f28-bdb9-a3434923547d)

![image](https://github.com/user-attachments/assets/02bfcbed-bb32-4728-8ea2-e8415708715b)

![image](https://github.com/user-attachments/assets/9e3abf93-89ff-4924-beb2-2e39bc5e4d3b)

![image](https://github.com/user-attachments/assets/a6a5e57c-498b-497a-9abc-e0ab1e1abf02)

![image](https://github.com/user-attachments/assets/4845bb4a-7e06-4452-992c-4def4261ad6a)

![image](https://github.com/user-attachments/assets/0865885c-321c-4624-9801-e384b0a1210e)

![image](https://github.com/user-attachments/assets/a0390d9a-8e8b-4662-a96b-55cb9cf1baee)

![image](https://github.com/user-attachments/assets/25b2e491-3bbd-4fec-996e-8fe3f203ef57)

## Checklist
